src/callbacks.c

```diff
@@ -157,6 +157,27 @@ void on_close_all1_activate(GtkMenuItem *menuitem, gpointer user_data)
 }
 
 
+void on_close_allafter1_activate(GtkMenuItem *menuitem, gpointer user_data)
+{
+	guint i;
+	GeanyDocument *cur_doc = user_data;
+
+	if (cur_doc == NULL)
+		cur_doc = document_get_current();
```

Comment:
The way you connect the callback it is unexpected to get a NULL user_data. Therefore this should be an error condition (catched by an assert (g_return_if_fail()) instead of choosing the current doc.